xyz-hub-service/src/main/java/com/here/xyz/httpconnector/config/AwsSecretManagerClient.java

```diff
@@ -0,0 +1,66 @@
+package com.here.xyz.httpconnector.config;
+
+import com.amazonaws.auth.AWSCredentials;
+import com.amazonaws.auth.AWSStaticCredentialsProvider;
+import com.amazonaws.auth.BasicAWSCredentials;
+import com.amazonaws.client.builder.AwsClientBuilder;
+import com.amazonaws.services.secretsmanager.AWSSecretsManager;
+import com.amazonaws.services.secretsmanager.AWSSecretsManagerClientBuilder;
+import com.amazonaws.services.secretsmanager.model.GetSecretValueRequest;
+import com.amazonaws.services.secretsmanager.model.GetSecretValueResult;
+import com.here.xyz.httpconnector.CService;
+import com.here.xyz.hub.util.ARN;
+import io.vertx.core.json.JsonObject;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+public class AwsSecretManagerClient {
+    private static final Logger logger = LogManager.getLogger();
+
+    private final AWSSecretsManager client;
+
+    public AwsSecretManagerClient() {
+        AWSSecretsManagerClientBuilder builder = AWSSecretsManagerClientBuilder.standard();
+
+        if(isLocal()){
+            builder.withEndpointConfiguration(new AwsClientBuilder.EndpointConfiguration(
+                            CService.configuration.LOCALSTACK_ENDPOINT, CService.configuration.JOBS_REGION))
+                    .withCredentials(new AWSStaticCredentialsProvider(new BasicAWSCredentials("localstack", "localstack")));
+        }
+
+        this.client = builder.build();
+    }
+
+    public String getSecret(ARN secretArn) {
+        String resource = secretArn.getResourceWithoutType();
+        String secretName = resource.substring(0, resource.lastIndexOf('-'));
```

Comment:
String manipulation of ARN seems to not be necessary as GetValueRequest#withSecretId() can also take an ARN directly.

See according AWS docs: https://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/services/secretsmanager/model/GetSecretValueRequest.html#withSecretId-java.lang.String-